sunpy/map/sources/yohkoh.py

```diff
@@ -56,15 +56,12 @@ def __init__(self, data, header, **kwargs):
         # use simple trigonometry to calculate the distance of the center of
         # the Sun from the spacecraft.  Note that the small angle approximation
         # is used, and the solar radius stored in SXT FITS files is in arcseconds.
-        self.meta['dsun_apparent'] = constants.au
         if 'solar_r' in self.meta:
             self.meta['dsun_apparent'] = constants.radius/(np.deg2rad(self.meta['solar_r']/3600.0))
+        else:
+            self._fix_and_warn('dsun_apparent', constants.au)
```

Comment:
I don't think we need this to throw either, this looks like we are merrily injecting our own header value.

Reply:
Ugh, I don't like this Yohkoh handling.  We shouldn't add a new item to the dictionary for this.

This comment appears to suggest that the Yohkoh header has the wrong information in the FITS keyword `DSUN_OBS`, which is why a custom header keyword is created:
https://github.com/sunpy/sunpy/blob/db314114f0148d16610dda10f18ea0876942cd61/sunpy/map/sources/yohkoh.py#L52-L54

But, the `dsun` property is written as if `DSUN_OBS` is accurate if present (i.e, `meta['dsun_apparent']` may not actually be used):
https://github.com/sunpy/sunpy/blob/db314114f0148d16610dda10f18ea0876942cd61/sunpy/map/sources/yohkoh.py#L67

So, is `DSUN_OBS` inaccurate or is it not defined?

It seems to me that:

- If `DSUN_OBS` is wrong, it should be fixed and the warning raised upon fixing.
- If `DSUN_OBS` is just sometimes missing, the warning should be raised on access in the `dsun` property.
- Either way, there's no reason to add a new item to the dictionary.
- The fallback should never have a path to a fixed `constants.au`, but rather we should be falling back on an Earth observer.

Reply:
Might be best to open this in a new issue? I'll revert the changes here.

Reply:
Created: #4395